### PR TITLE
Addresses issue #605 -- Fetch JSON for US/VA_GOV

### LINKED ISSUE #605: fetch us/va_gov
[learn our pipeline: fetch](https://github.com/CAVaccineInventory/vaccine-feed-ingest/wiki/Runner-pipeline-stages#fetch)

Fetch data from: https://www.va.gov/find-locations/?address=20002&bounds%5B%5D%5B%5D=-123.02&bounds%5B%5D%5B%5D=37.08&bounds%5B%5D%5B%5D=-121.52&bounds%5B%5D%5B%5D=38.58&bounds%5B%5D%5B%5D%5B%5D=-123.02&bounds%5B%5D%5B%5D%5B%5D=37.08&bounds%5B%5D%5B%5D%5B%5D=-121.52&bounds%5B%5D%5B%5D%5B%5D=38.58&bounds%5B%5D%5B%5D%5B%5D%5B%5D=-123.02&bounds%5B%5D%5B%5D%5B%5D%5B%5D=37.08&bounds%5B%5D%5B%5D%5B%5D%5B%5D=-121.52&bounds%5B%5D%5B%5D%5B%5D%5B%5D=38.58&context=Washington%2C%20District%20of%20Columbia%2020002%2C%20United%20States&facilityType=health&latitude=38.89&longitude=-77.01&page=1&radius=10&serviceType=Covid19Vaccine&bounds%5B%5D=-77.76&bounds%5B%5D=38.14&bounds%5B%5D=-76.26&bounds%5B%5D=39.64

Be sure to update request parameters to fetch data nationally, but to continue to filter for `Covid19Vaccine` only.

Put your script in a file named: `us/va_gov/fetch.py` (or `fetch.sh`, etc.)

Store it (without processing) in a new file created in the directory passed as the first argument (`sys.argv[1]`).

Check the wiki to learn more about the purpose of the fetch stage and how to get set up for development!

### Tips

1. While working on your code, run it at any point:
    ```sh
    poetry run vaccine-feed-ingest fetch us/va_gov
    ```

### PR DESCRIPTION
**Pull request by davekasten**
# Fetch JSON for US/VA_GOV -->

| Key Details |
|-|
| Resolves #605 |
State: US
Site: va_gov

## Notes

Created as per spec from VA's API documentation. 
Requires a key that can be self-serve created on VA's website (link in code comments).

## Data sample
{"type":"FeatureCollection","features":[{"type":"Feature","geometry":{"type":"Point","coordinates":[-71.03613513,42.34459497]},"properties":{"id":"vc_0101V","name":"Boston Vet Center","facility_type":"vet_center","classification":null,"website":null,"address":{"mailing":{},"physical":{"zip":"02210-2303","city":"Boston","state":"MA","address_1":"7 Drydock Avenue","address_2":"Suite 2070","address_3":null}},"phone":{"fax":"617-261-3547","main":"857-203-6461"},"hours":{"friday":"830AM-430PM","monday":"830AM-700PM","sunday":"Closed","tuesday":"830AM-700PM","saturday":"Closed","thursday":"830AM-600PM","wednesday":"830AM-600PM"},"operational_hours_special_instructions":"Expanded or Nontraditional hours are available for some services on a routine and or requested basis. Please call our main phone number for details. |","services":{},"satisfaction":{},"wait_times":{},"mobile":false,"active_status":"A","operating_status":{"code":"NORMAL"},"detailed_services":null,"visn":"1"}},{"type":"Feature","geometry":{"type":"Point","coordinates":[-74.07389209,40.78911661]},"properties":{"id":"vc_0102V","name":"Secaucus Vet Center","facility_type":"vet_center","classification":null,"website":null,"address":{"mailing":{},"physical":{"zip":"07094-2302","city":"Secaucus","state":"NJ","address_1":"110A Meadowlands Parkway","address_2":"Suite 102","address_3":null}},"phone":{"fax":"201-223-7707","main":"201-223-7787"},"hours":{"friday":"800AM-430PM","monday":"800AM-430PM","sunday":"Closed","tuesday":"800AM-430PM","saturday":"Closed","thursday":"800AM-430PM","wednesday":"800AM-430PM"},"operational_hours_special_instructions":"Expanded or Nontraditional hours are available for some services on a routine and or requested basis. Please call our main phone number for details. | Vet Center after hours assistance is available by calling 1-877-WAR-VETS (1-877-927-8387).","services":{},"satisfaction":{},"wait_times":{},"mobile":false,"active_status":"A","operating_status":{"code":"NORMAL"},"detailed_services":null,"visn":"2"}},
```json

```

## Before Opening a PR
- [ Y] I tested this using the CLI (e.g., `poetry run vaccine-feed-ingest <state>/<site>`)
- [ Y] I ran auto-formatting: `poetry run tox -e lint-fix`

### CODE REVIEW

**Comment by nuchi** [comment]
I was able to get nationwide (including at least Puerto Rico, not sure about other territories?) data via [this link](https://api.va.gov/v1/facilities/va?bbox[]=-180&bbox[]=-90&bbox[]=180&bbox[]=90&type=health&services[]=Covid19Vaccine&page=1&per_page=2000&radius=25000&latitude=37.408123149415275&longitude=-93.14343299172322), and without needing to authenticate. @davekasten does that duplicate the info you're able to get?

**Comment by davekasten** [comment]
> I was able to get nationwide (including at least Puerto Rico, not sure about other territories?) data via [this link](https://api.va.gov/v1/facilities/va?bbox%5B%5D=-180&bbox%5B%5D=-90&bbox%5B%5D=180&bbox%5B%5D=90&type=health&services%5B%5D=Covid19Vaccine&page=1&per_page=2000&radius=25000&latitude=37.408123149415275&longitude=-93.14343299172322), and without needing to authenticate. @davekasten does that duplicate the info you're able to get?

Interesting idea to hack around it with the distance on the API that way! Hard to do a true 1-to-1 comparison (I'm sure there's some way to do it by parsing through the file but that's beyond my skills) but did several spot checks. 

I think your option is duplicative for _all sites with vaccine_.  Your version is, I think, better in that it should not include sites w/o vaccine (whereas mine does), which makes data cleaning this in the later steps easier since we don't have to pull the non-vax sites out.  Given that, think we should proceed with the approach you suggest!

What's the right next steps here / right etiquette? (Sorry, I've made very limited use of Github in the past so not sure right way of updating something that's already in a PR)

**Comment by nuchi** [comment]
>  Given that, think we should proceed with the approach you suggest!
>
> What's the right next steps here / right etiquette? (Sorry, I've made very limited use of Github in the past so not sure right way of updating something that's already in a PR)

Awesome! I think the fastest thing (and in this circumstance "fastest" is in my opinion the metric to optimize for) would be for you to switch the link locally, commit, and then push the new commit to your branch `davekasten:main`. The PR will automatically update to include that. That'd probably be faster than me opening a PR!

**Comment by davekasten** [comment]
> Awesome! I think the fastest thing (and in this circumstance "fastest" is in my opinion the metric to optimize for) would be for you to switch the link locally, commit, and then push the new commit to your branch `davekasten:main`. The PR will automatically update to include that. That'd probably be faster than me opening a PR!

great -- updated!  Also updated the comments in the script to note (as per convo w/Jesse on Fri on Discord) that the underlying VA data source might not include some VA pop-up vaccination events (e.g. Veterans of Foreign Wars halls)

File: vaccine_feed_ingest/runners/us/va_gov/fetch.sh [hunk 1]
```diff
@@ -10,14 +10,5 @@ else
...
-
-
-
-##takes an API key that you can self-service create from https://developer.va.gov/apply -- or I can provide
-##this secret out-of-band via Discord or whatever
-
-##Note: Sandbox appears to be appropriate environment for read-only access; it appears prod is only for authorized developers that are building things that 
-##can change state on the VA server side.  Data is accurate as far as I can tell against what's available in real time on VA website.
+# Uses official VA API for searching for locations.  NOTE: underlying data source may not include some VA pop up vaccination events at smaller VA clinics or locations like VFW halls.
+(cd "$output_dir" && curl -X GET "https://api.va.gov/v1/facilities/va?bbox%5B%5D=-180&bbox%5B%5D=-90&bbox%5B%5D=180&bbox%5B%5D=90&type=health&services%5B%5D=Covid19Vaccine&page=1&per_page=2000&radius=25000&latitude=37.408123149415275&longitude=-93.14343299172322" --create-dirs -o 'va_gov.json')
```
Comment (eliblock) [inline vaccine_feed_ingest/runners/us/va_gov/fetch.sh:14]: It looks like this currently returns `442` sites, which is well under the `2,000` site page size.

**Review by eliblock — CHANGES_REQUESTED** [review]
Thank you @davekasten and @nuchi! I think that this site should be a good fit for the [shared ndjson parser](https://github.com/CAVaccineInventory/vaccine-feed-ingest/wiki/JSON-to-NDJSON-Parser) to make the next step a bit easier!

One note: the script needs to be executable in order to run this via our tool. You can change this by running `chmod +x vaccine_feed_ingest/runners/us/va_gov/fetch.sh`.

File: vaccine_feed_ingest/runners/us/va_gov/fetch.sh (hunk 1 above)
Comment (davekasten) [reply vaccine_feed_ingest/runners/us/va_gov/fetch.sh:14]: > Thank you @davekasten and @nuchi! I think that this site should be a good fit for the [shared ndjson parser](https://github.com/CAVaccineInventory/vaccine-feed-ingest/wiki/JSON-to-NDJSON-Parser) to make the next step a bit easier!
> 
> One note: the script needs to be executable in order to run this via our tool. You can change this by running `chmod +x vaccine_feed_ingest/runners/us/va_gov/fetch.sh` and then committing the change.

Thanks -- I chmod'd the script and uploaded a new version.  Checking with ls -l returns that it has executable permissions, so I (think) it should be executable now

**Review by eliblock — APPROVED** [review]
Thank you for the further discussion offline!

**Pull request merged**